_posts/dusan_andric/2018-05-25-icgc-jupyter.md

```diff
@@ -0,0 +1,92 @@
+title:  "Jupyter in ICGC"
+breadcrumb: true
+author: dusan_andric
+date: 2018-05-25
+categories: dusan_andric
+tags:
+    - ICGC
+    - Collaboratory
+    - Jupyter
+    - Analysis
+    - Data Science
+    - Overture
+teaser:
+    info: The ICGC Data Portal gets a new analysis tool! 
+    image: dusan_andric/jupyter/jupyter.png
+header: 
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog
+---
+
+## Introduction
+
+After a successful soft launch, the ICGC DCC Software Engineering is proud to announce the availability of Jupyter notebooks as a new tool in the analysis toolbox within the ICGC Data Portal. 
+
+Users with DACO approval to the ICGC dataset will have the ability to login using their Google credentials. Once inside, users are presented with a familiar looking Jupyter Notebook environment with some value adds provided by us. 
+
+Our goal is to provide researchers and data scientists a way to programmatically explore the available ICGC data and get their hands dirty with some data science and analysis in a way that is sharable and reproducable. 
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/jupyter/jupyter_signin.png" />
+    <figcaption>The Welcome Page</figcaption>
+</figure>
+
+
+Users can get to Jupyter by either visiting the following URL directly in their browser: https://jupyterhub.cancercollaboratory.org/ or by navigating to the [analysis toolbox in the ICGC Data Portal](https://dcc.icgc.org/analysis)
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/jupyter/analysis_page.png" />
+    <figcaption>The Analysis Toolbox</figcaption>
+</figure>
+
+## Technology
+
+For a detailed description of all the moving pieces that went into this, please check out [Kevin](http://softeng.oicr.on.ca/blog/category/kevin_hartmann)'s excellent [blog post](http://softeng.oicr.on.ca/kevin_hartmann/2018/03/28/Drops-of-Jupyter/) on the technical aspects of deploying JupyterHub.
+
+The main driving project behind our ability to deploy this new feature is JupyterHub. JupyterHub itself is a bundle of three main technologies (tornado, node-http-proxy, jupyter) which allows notebook servers to be provisioned on a per user or session basis. 
+
+The other secret sauce is that we are using the docker spawner for launching new notebook servers using a docker image that we hand rolled ourselves to include the particular customizations and features that we want. JupyterHub and the docker containers all run inside the Cancer Genome Collaboratory cloud. 
+
+We've also extended the Authorization mechanism to perform the required DACO checks for access to the ICGC dataset in addition to generalizing the whole deployment using Ansible.
+
+The Ansible deployment of this solution, from the building of docker images to the launching of VMs, has been made available as [Jukebox](https://github.com/overture-stack/Jukebox) which is part of the [Overture software stack](http://www.overture.bio/).  
```

Comment:
"has been made available as Jukebox", I would rephrase into "has been made available as an Overture component called Jukebox" (with the corresponding links).